ntbs-service/Models/Entities/Notification.cs

```diff
@@ -127,10 +127,19 @@ public Notification()
                                        || NotificationStatus == NotificationStatus.Closed
                                        || NotificationStatus == NotificationStatus.Legacy;
 
-        [AssertThat(@"ShouldValidateFull && HasDeathEventForPostMortemCase", ErrorMessage = ValidationMessages.DeathEventRequiredForPostMortemCase)]
-        public bool HasDeathEventForPostMortemCase =>
-            ClinicalDetails.IsPostMortem != true || (TreatmentEvents != null && TreatmentEvents.Any(x =>
-                x.TreatmentEventTypeIsOutcome && x.TreatmentOutcome.TreatmentOutcomeType == TreatmentOutcomeType.Died));
+        [AssertThat(
+            @"ShouldValidateFull && (ClinicalDetails.IsPostMortem != true || IsPostMortemAndHasCorrectEvents)",
+            ErrorMessage = ValidationMessages.IsPostMortemButTreatmentEventsDoNotMatch)]
+        public bool IsPostMortemAndHasCorrectEvents =>
+            (ClinicalDetails?.IsPostMortem ?? false) && TreatmentEvents != null &&
+            (TreatmentEvents.Any(x => x.TreatmentEventTypeIsOutcome && x.TreatmentOutcome?.TreatmentOutcomeType == TreatmentOutcomeType.Died) &&
+             (TreatmentEvents.Count == 1 || (HasCorrectDiagnosisMadeEvent)));
```

Comment:
This may just be a matter of personal taste, but I find expressions like this easier to follow when each condition is on a new line, and the operator comes at the start of the line rather than the end. So something like this here:
```
        public bool IsPostMortemAndHasCorrectEvents =>
            ClinicalDetails?.IsPostMortem == true
                && TreatmentEvents != null
                && TreatmentEvents.Any(x => x.TreatmentEventTypeIsOutcome
                    && x.TreatmentOutcome?.TreatmentOutcomeType == TreatmentOutcomeType.Died)
                && (TreatmentEvents.Count == 1 || HasCorrectDiagnosisMadeEvent);
```